### DISCUSSION TITLE
The way to get all members with certain role.

### DISCUSSION THREAD

**Discussion by sanovskiy** [discord-php/DiscordPHP#1166] [Q&A] [answered]
I don't know if the Discord API has the functionality to get users with a specific role. Retrieving all users and then iterating over them to find the right role is not a very optimal solution.
\Discord\Parts\Guild\Role doesn't have Members repository. So is there any way to fetch users with specific role?

**Comment by key2peace** [comment, marked as answer]
There is ->filter() which pretty much does the same, just looks easier to use
```php
$roleholders = $guild->members->filter(function($member) use ($role) { return $member->roles->has($role->id); });
```
This should return a Collection of members with a specified role.

**Reply by sanovskiy** [reply]
It returns an empty collection. I think it's because not all users are cached. But there is at least one user with this role on the server.

UPD: I found out why it was empty. I had to add 'loadAllMembers' => true to Discort constructor.

**Comment by pkSourceCode** [comment]
How to get all members under the PRESENCE_UPDATE event plz help..
